apps/minifront/src/components/swap/swap-form/token-swap-input.tsx

```diff
@@ -143,21 +138,41 @@ export const TokenSwapInput = () => {
             </div>
           )}
 
-          <div className='flex h-full flex-col gap-2'>
-            <BalanceSelector value={assetIn} onChange={setAssetIn} balances={balancesResponses} />
-            {assetIn?.balanceView && (
-              <BalanceValueView valueView={assetIn.balanceView} onClick={setInputToBalanceMax} />
-            )}
-          </div>
-
-          <div className='flex flex-col gap-2 pt-2'>
-            <ArrowRight size={16} className='text-muted-foreground' />
-          </div>
-
-          <div className='flex h-full flex-col gap-2'>
-            <AssetSelector assets={swappableAssets} value={assetOut} onChange={setAssetOut} />
-            {assetOut && <BalanceValueView valueView={assetOutBalance} />}
-          </div>
+          <FadeIn condition={!!balancesResponses.data && !!swappableAssets.data}>
```

Comment:
This fades in the input/output selectors once the necessary data has loaded:

![screencap](https://github.com/penumbra-zone/web/assets/1121544/bf33baf4-b41d-4260-a8f4-d1c2612837a5)

Reply:
Can we handle the error case as well? Maybe some red text with a loading error message?

Reply:
done